Add a GOVERNANCE.md document

criccomini opened the pull request:
Adding a governance document based on [Vespa's](https://github.com/vespa-engine/vespa/blob/master/GOVERNANCE.md).

I tweaked the following things:

- Make all committers board members for now.
- Add reference to slatedb-pulumi for the official committer list.
- Cleaned some grammar.
- Added a carveout to allow committers to merge small PRs without approval.

rodesai commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,54 @@
+
+# The SlateDB Board
+
+The __SlateDB Board__ has final authority over all decisions regarding the SlateDB project,
```
should we specify how we decide (e.g. simple majority)?

rodesai commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,54 @@
+
+To become a SlateDB Committer, start making PRs that are approved. When a sufficient
+trust is built through working on SlateDB this way, any current SlateDB Committer can nominate
+you to the board for the committer role.
```
how should we decide on adding a committer? the board votes (I think this is pretty typical)?

rodesai replied:
nvm see this is addressed below

rodesai commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,54 @@
+trust is built through working on SlateDB this way, any current SlateDB Committer can nominate
+you to the board for the committer role.
+
+Every pull request should be approved by a SlateDB Committer other than the author of the
```
I think we should have a higher standard for RFCs - maybe 2 committer approvals and no objecting votes from committers?

criccomini commented:
Updated GOVERNANCE.md and RELEASE.md with:

- Exclude slatedb-go
- Add releases to governance
- Exapand release doc with cadence, proposal, and approval
- Add RFC to governance
- Split board from committers

paulgb reviewed (approved):
LGTM!
